### PR TITLE
new files for MCPL

### PR DESCRIPTION
**Pull request by grzanka**
I've fixed the PDG codes in the `51-phsp-export` branch of SHIELD-HIT12A source code. This change needs to be reflected in the testing files for pymchelper. Previously allmost all the particles were neutrons, while we expected protons.

This PR includes as well solid refactoring of tests suite (moving from unittest to pytest framework).

Concatenation for the MCPL data object was also implemented, which should enable dealing with concatenating large files using `convertmc --many *.bdo`

### CODE REVIEW

File: pymchelper/input_output.py [hunk 1]
```diff
@@ -130,7 +138,7 @@ def fromfilelist(input_file_list, error=ErrorEstimate.stderr, nan: bool = True):
     return result
 
 
-def frompattern(pattern, error=ErrorEstimate.stderr, nan=True):
+def frompattern(pattern: str, error: ErrorEstimate = ErrorEstimate.stderr, nan: bool = True):
```
Comment (Bogdan2423) [inline pymchelper/input_output.py:141]: Add type hint for return type

File: pymchelper/input_output.py [hunk 2]
```diff
@@ -176,12 +184,7 @@ def convertfromlist(filelist, error, nan, outputdir, converter_name, options, ou
     return status
 
 
-def convertfrompattern(pattern,
-                       outputdir,
-                       converter_name,
-                       options,
-                       error=ErrorEstimate.stderr,
-                       nan: bool = True):
+def convertfrompattern(pattern, outputdir, converter_name, options, error=ErrorEstimate.stderr, nan: bool = True):
```
Comment (Bogdan2423) [inline pymchelper/input_output.py:187]: Add type hints

File: tests/conftest.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+                # move to installation path
+                local_file_path = Path(unpacking_dir) / member.filename
+                destination_file_path = installation_dir / member_name
+                logging.info("Moving %s to %s", local_file_path, installation_dir)
+                # move file from temporary directory to installation path using shutils
+                if not destination_file_path.exists():
+                    shutil.move(local_file_path, destination_file_path)
+
+
+def download_shieldhit_demo_version(installation_dir):
```
Comment (Bogdan2423) [inline tests/conftest.py:50]: Add type hint

File: tests/test_averaging.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    yield main_dir / "res" / "shieldhit" / "generated" / "many" / "msh"
+
+
+@pytest.fixture(scope='function')
+def shieldhit_multiple_result_pattern() -> Generator[str, None, None]:
+    """Return SHIELD-HIT12A result files as glob generator"""
+    yield "aen_x_al0*.bdo"
+
+
+def test_averaging(shieldhit_multiple_result_pattern, shieldhit_multiple_result_directory):
```
Comment (Bogdan2423) [inline tests/test_averaging.py:26]: Add type hints

**Review by nbassler — APPROVED** [review]
tested, works.

File: tests/conftest.py [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+import tarfile
+import tempfile
+from pathlib import Path
+from typing import Generator
+import zipfile
+
+import pytest
+import requests
+
+sh12a_ver = '1.0.1'
```
[the remark this answers is not in the document]
Comment (grzanka) [reply tests/conftest.py:13]: I will update these tests while 1.1.0 is released. Pytest right now
 - automatically downloads demo version of SH12A 1.0.1
 - runs shieldhit binary to generate few BDO files (excluding MCPL scorers)
 - checks if shieldhit runner (`runmc` command) works properly
 - checks if the BDO files are parsed properly

Tests of MCPL files reading (`test_mcpl.py`) are based on pregenerated files.

Once new released is made I can add MCPL to automatic generation suite.

**Pull request merged**